sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -585,11 +585,23 @@ protected void initialize(SchedulerDriver driver) throws InterruptedException {
     }
 
     private void killUnneededTasks(Set<String> taskToDeployNames) {
-        Set<Protos.TaskID> taskIds = stateStore.fetchTasks().stream()
+        Set<Protos.TaskInfo> taskInfos = stateStore.fetchTasks().stream()
                 .filter(taskInfo -> !taskToDeployNames.contains(taskInfo.getName()))
+                .collect(Collectors.toSet());
+
+        Set<Protos.TaskID> taskIds = taskInfos.stream()
                 .map(taskInfo -> taskInfo.getTaskId())
                 .collect(Collectors.toSet());
 
+        // Clear the TaskIDs from the TaskInfos so we drop all future TaskStatus Messages
+        Set<Protos.TaskInfo> cleanedTaskInfos = taskInfos.stream()
+                .map(taskInfo -> taskInfo.toBuilder())
+                .map(builder -> builder.setTaskId(Protos.TaskID.newBuilder().setValue("")).build())
+                .collect(Collectors.toSet());
+
+        cleanedTaskInfos.forEach(taskInfo -> stateStore.clearTask(taskInfo.getName()));
```

Comment:
No task reconciliation would occur, because the TaskID is gone, so there's nothing to reconcile with.  The way out of this situation is to manually kill tasks.

1. The modifications to StateStore say this thing has never been launched.
1. The modifications which can be done in the plan to disable a task say don't deploy this task.

If you crash the scheduler in the middle we won't get around to killing tasks because lost the TaskID.  So you've have to manually kill it.  You could `pod restart` and that should work.